### PR TITLE
Add support for application/x-pem-file MIME type

### PR DESCRIPTION
**Pull request by robin-nitrokey**
This patch adds support for serializing and deserializing data with the application/x-pem-file MIME type as plain text.

Fixes: https://github.com/openapi-json-schema-tools/openapi-json-schema-generator/issues/244

Minimal example:

```
openapi: 3.0.0
info:
  title: Sample API
  version: 0.1.0
servers:
  - url: http://api.example.com/v1
paths:
  /:
    get:
      summary: dummy
      requestBody:
        content:
          application/x-pem-file:
            schema:
              type: string
      responses:
        '200':
          description: dummy
          content:
            application/x-pem-file:
              schema: 
                type: string
```

Real world example: https://github.com/Nitrokey/nethsm-sdk-py/pull/65

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapi-json-schema-tools/openapi-json-schema-generator/blob/master/docs/contributing.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-json-schema-generator#14---build-projects) and update samples:
  ```
  mvn clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/python*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).

### CODE REVIEW

**Comment by robin-nitrokey** [comment]
Not sure why the CI fails.  I’ve ran the scripts mentioned in the PR template to update the samples but it looks like it did not work.

File: src/main/resources/python/api_client.hbs [hunk 1]
```diff
@@ -1404,7 +1406,7 @@ class RequestBody(StyleFormSerializer, JSONDetector):
             if isinstance(cast_in_data, (schemas.FileIO, bytes)):
                 raise ValueError(f"Invalid input data type. Data must be int/float/str/bool/None/tuple/immutabledict and it was type {type(cast_in_data)}")
             return cls.__serialize_json(cast_in_data)
-        elif content_type == 'text/plain':
+        elif content_type in ['text/plain', 'application/x-pem-file']:
```
Comment (spacether) [inline src/main/resources/python/api_client.hbs:1409]: How about using a set here and making it a class constant at the top of the class?
`__plain_txt_content_types = {'text/plain', 'application/x-pem-file'}`
`elif content_type in cls.__plain_txt_content_types:`

**Comment by spacether** [comment]
Thank you for the PR. Can you please run these steps and then commit the changes?
```
mvn clean package 
./bin/generate-samples.sh
```
CI is failing because the repo samples have changes when they are regenerated with your fixes.

File: src/main/resources/python/api_client.hbs (hunk 1 above)
Comment (robin-nitrokey) [reply src/main/resources/python/api_client.hbs]: Good point, I changed it.

**Comment by robin-nitrokey** [comment]
Oh, I ran the script, added the files but forgot to amend the commit. :facepalm: Should be fixed now.

**Comment by robin-nitrokey** [comment]
… and rebased onto master.

**Comment by robin-nitrokey** [comment]
… and finally fixed the CI.

**Comment by spacether** [comment]
Thank you for making those changes. The PR as-is has no tests verifying that it works. If we leave it as-is a future PR could break this functionality without generating a test error signal.
How about
- adding a route in [the petstore spec](https://github.com/openapi-json-schema-tools/openapi-json-schema-generator/blob/master/src/test/resources/3_0/python/petstore_customized.yaml) which sends and receives your needed content type
- a test in [this directory](https://github.com/openapi-json-schema-tools/openapi-json-schema-generator/tree/master/samples/client/petstore/python/tests_manual/test_paths) verifying that sending and receiving via  that route works? [Similar-to test here](https://github.com/openapi-json-schema-tools/openapi-json-schema-generator/blob/master/samples/client/petstore/python/tests_manual/test_paths/test_fake/test_post.py#L35)

**Review by spacether — APPROVED** [review]
Thank you for your PR. I never heard back from you about adding a test so I will add one after merging it.

**Comment by robin-nitrokey** [comment]
@spacether Thank you!  I started working on the test but got stuck figuring out the details of the test setup, so I very much appreciate you adding the test.

**Pull request merged**